pyteal/ast/seq.py

```diff
@@ -51,7 +51,17 @@ def __init__(self, *exprs):
             if not isinstance(expr, Expr):
                 raise TealInputError("{} is not a pyteal expression.".format(expr))
             if i + 1 < len(exprs):
-                require_type(expr, TealType.none)
+                try:
+                    require_type(expr, TealType.none)
+                except TealTypeError as e:
+                    e.__class__ = TealSeqError
+                    e.message = "{} must have a return type of TealType.none. ".format(
+                        expr
+                    )
+                    e.message += (
+                        "Only the last entry of a Seq array can have a return value."
+                    )
```

Comment:
This PR seems like a good change but I dislike this method of modifying the error object. Can you either:
* Create a new `TealSeqError` here and raise it
* Use a new function instead of `require_type` which raises the right error in the first place